src/components/TextInput/TextInputLabel/index.js

```diff
@@ -1,31 +1,32 @@
-import React, {PureComponent} from 'react';
+import React, {useRef, useEffect} from 'react';
 import {Animated} from 'react-native';
 import styles from '../../../styles/styles';
 import {propTypes, defaultProps} from './TextInputLabelPropTypes';
 import CONST from '../../../CONST';
 
-class TextInputLabel extends PureComponent {
-    componentDidMount() {
-        if (!this.props.for) {
+function TextInputLabel(props) {
+    const labelRef = useRef(null);
+
+    useEffect(() => {
+        if (!props.for || !labelRef.current) {
             return;
         }
-        this.label.setAttribute('for', this.props.for);
-    }
+        labelRef.current.setAttribute('for', props.for);
+    }, [props.for]);
```

Comment:
This changes the behavior. Originally this was called only once on `componentDidMount`, no matter `for` prop changes dynamically.
So to be consistent, it makes sense to use empty array as dependency.